### DISCUSSION TITLE
how to make mobile-first website

### DISCUSSION THREAD

**Discussion by rajeshtva** [quasarframework/quasar#10300] [General - Components / Directives / etc]
i am working on website project. which is required to be mobile-first. but when i search _mobile-first_ in official documentation,  then i don't find any mention of this. so my questions are

 - how can i design my website like this?. 
 - do i need to use media query or **$q.is.platform** and how can i use these two?
 - do i need to define seperate block for different kinds of devices for this?

 i am a beginner in medium knowledge of vue framework but poor knowledge of ui/ux and appropriate css.

**Comment by MilosPaunovic** [comment]
I would suggest you to take a look at [Quasar spacing]( https://quasar.dev/style/spacing) docs page.
This approach allows you to create one layout and adjust it for different viewports.

**Comment by ontwikkelfabriek** [comment]
@je-rajesh 

> how can i design my website like this?.

Quasar has a responsive layout system ( using flexbox). It will respond to screen sizes.  Just be sure to read the **flexbox** and **layout** part of quasar docs. 

https://quasar.dev/layout-builder

> do i need to use media query or $q.is.platform and how can i use these two?

No you don't need that. 

> do i need to define seperate block for different kinds of devices for this?

No , you can but you don't have to.

To do mobile first design with Quasar, develop your app with chrome/firefox put in **mobile** view mode( see dev tools). This combined with the hot module reload will give you a nice dev experience. 

Tip , read all of the Quasar( and Vue) docs at least once from top to bottom.
